src/probnum/kernels/_matern.py

```diff
@@ -0,0 +1,115 @@
+"""Matern kernel."""
+
+from typing import Optional
+
+import numpy as np
+import scipy.spatial.distance
+import scipy.special
+
+import probnum.utils as _utils
+from probnum.type import IntArgType, ScalarArgType
+
+from ._kernel import Kernel
+
+_InputType = np.ndarray
+
+
+class Matern(Kernel[_InputType]):
+    """Matern kernel.
+
+    Covariance function defined by :math:`k(x_0, x_1) = \\frac{1}{\\Gamma(\\nu)2^{
+    \\nu-1}}\\big(\\frac{\\sqrt{2\\nu}}{l} \\lVert x_0 , x_1\\rVert \\big)^\\nu
+    K_\\nu\\big(\\frac{\\sqrt{2\\nu}}{l} \\lVert x_0 , x_1 \\rVert \\big)`, where
+    :math:`K_\\nu` is a modified Bessel function. The Matern
+    kernel generalizes the :class:`~probnum.kernels.ExpQuad` kernel
+    via its additional parameter :math:`\\nu` controlling the smoothness of the
+    function. For :math:`\\nu \\rightarrow \\infty` the Matern kernel converges to
+    the :class:`~probnum.kernels.ExpQuad` kernel. A Gaussian process
+    with Matern covariance function is :math:`\\lceil \\nu \\rceil - 1` times
+    differentiable.
+
+    Parameters
+    ----------
+    input_dim :
+        Input dimension of the kernel.
+    lengthscale :
+        Lengthscale of the kernel. Describes the input scale on which the process
+        varies.
+    nu :
+        Hyperparameter controlling differentiability.
```

Comment:
is nu a good name? Wouldn't maybe `reg` (regularity) or `smoothness` or so be more descriptive? FYI: in state space models, the respective value of `nu` is referred to as the order (if I remember correctly, `ordint` for order of integration).

Reply:
I think this is one of those edge cases where the mathematical symbol is so well known, that it is more descriptive than choosing another name. In particular `smoothness` is not quite correct, because of the added 0.5.

Reply:
> I think this is one of those edge cases where the mathematical symbol is so well known, that it is more descriptive than choosing another name

Is it though? gpytorch does it like this, and wikipedia writes it, but I have already seen many instances where people do not use \nu (e.g. in numerics, people usually use "m").  Though in any case, feel free to go ahead with this, I merely wanted to propose alternatives :)